app/web/features/userQueries/useLiteUsers.test.tsx

```diff
@@ -0,0 +1,283 @@
+  const client = new QueryClient({
+    defaultOptions: {
+      queries: {
+        retry: false,
+      },
+    },
+  });
+
+  return <QueryClientProvider client={client}>{children}</QueryClientProvider>;
+};
+
+describe("useLiteUsers & useLiteUser", () => {
+  describe("useLiteUsers - nonCached", () => {
+    const mockGetLiteUsers = service.user.getLiteUsers as jest.Mock;
+
+    const ids = [1, 2, 3, 4];
+
+    beforeEach(() => {
+      mockGetLiteUsers.mockResolvedValue(getLiteUsers(ids));
+    });
+
+    it("Should return loading state correctly", async () => {
+      const { result, waitFor } = renderHook(() => useLiteUsers(ids), {
+        wrapper,
+      });
+
+      expect(result.current.isLoading).toBe(true);
+      expect(result.current.data).toBeUndefined();
+      await waitFor(() => expect(result.current.isLoading).toBe(false));
+    });
+
+    it("Should return users map when loading is complete", async () => {
+      const { result, waitForNextUpdate } = renderHook(
+        () => useLiteUsers(ids),
+        {
+          wrapper,
+        }
+      );
+
+      await waitForNextUpdate();
+      expect(mockGetLiteUsers).toHaveBeenCalledTimes(1);
+
+      expect(result.current).toEqual(
+        expect.objectContaining({
+          data: new Map(mockLiteUsers.map((user) => [user.userId, user])),
+          error: null,
+          isError: false,
+          isFetching: false,
+          isLoading: false,
+          isRefetching: false,
+        })
+      );
+    });
+
+    it("Should filter out undefined ids", async () => {
+      const idsWithUndefined = [1, undefined, 3];
+
+      const { result, waitFor } = renderHook(
+        () => useLiteUsers(idsWithUndefined),
+        { wrapper }
+      );
+
+      await waitFor(() => expect(result.current.isLoading).toBe(false));
+
+      expect(mockGetLiteUsers).toHaveBeenCalledWith([1, 3]);
+    });
+
+    it("Should return an error if the query fails", async () => {
+      mockConsoleError();
+      const error = new RpcError(500, "Some error", {});
+      mockGetLiteUsers.mockRejectedValue(error);
+
+      const { result, waitForNextUpdate } = renderHook(
+        () => useLiteUsers([1, 2, 3]),
+        { wrapper }
+      );
+
+      await waitForNextUpdate();
+
+      expect(result.current.isLoading).toBe(false);
+      expect(result.current.error).toBe(error);
+      expect(result.current.data).toBeUndefined();
+    });
+
+    it("Should not run the query when ids is an empty array", async () => {
+      const ids: number[] = [];
+
+      renderHook(() => useLiteUsers(ids), { wrapper });
+
+      // Ensure the query function was NOT called
+      expect(mockGetLiteUsers).not.toHaveBeenCalled();
+    });
+
+    it("Should not run the query function when ids is undefined", () => {
+      const ids: (number | undefined)[] = [undefined];
+
+      renderHook(() => useLiteUsers(ids), { wrapper });
+
+      expect(mockGetLiteUsers).not.toHaveBeenCalled();
```

Comment:
If someone screw something up in the future and this IS being called by accident, wouldn't this not catch the problem since we're not awaiting for next update? i.e. you could write the same logic in the `Should return loading state correctly` test in the first part and this would still pass (at least from my understanding of reading the code)

Reply:
Oh wow good catch! I've added in a waitFor until `isLoading` is false!